### ISSUE TITLE
Error invoking bsh method: eval	Sourced file: inline evaluation of:

### ISSUE THREAD

**Issue by shreyaa1998** [apache/jmeter#5767]
### Expected behavior

Response message:org.apache.jorphan.util.JMeterException: Error invoking bsh method: eval	Sourced file: inline evaluation of: ``BufferedReader fileReader = new BufferedReader (new fileReader(vars.get("path")) . . . '' : Typed variable declaration : Unknown class: fileReader

Getting an error "Unknown class: fileReader" while trying to read a .txt file. Do we have a workaround? I am searching a method to read a specific line from a .txt text file and display that specific line in Jmeter console.

I used the below code, to read the 4th line:

BufferedReader fileReader = new BufferedReader (new fileReader(vars.get("path")));
int counter=4;
content = fileReader.readLine();
while ((nextLine = fileReader.readLine())!=null)
{
	content = content + "\n" + nextLine;
	counter++;
}
vars.put("content",content);

x=vars.get("content")

### Steps to reproduce the problem

Using a notepad and saving it .txt extension

### JMeter Version

5.3

**Comment by FSchumacher** [comment]
There are a few problems with the code.

First, Java classes are case sensitive and are starting with an upper case character. So `fileReader` is not a class name. The correct class name would be `FileReader`. (The variable name on the left side of the equation can be left as it is)
Second, you are not reading the fourth line, only. You are reading the complete content. You would have to add an if clause to capture the fourth line, only.
Third, I would use Groovy instead of Beanshell. It supports more modern Java features and a lot of cool stuff on top. Plus it is probably faster than Beanshell most of the time.

In Groovy, you could shorten this to
```groovy
vars.put("content") = new File(vars.get("path")).readLines()[3]
```
I used `3` for the fourth line, as the first line is at position zero.
